staging/src/k8s.io/client-go/dynamic/fake/simple.go

```diff
@@ -35,7 +35,9 @@ import (
 
 func NewSimpleDynamicClient(scheme *runtime.Scheme, objects ...runtime.Object) *FakeDynamicClient {
 	// In order to use List with this client, you have to have the v1.List registered in your scheme. Neat thing though
-	// it does NOT have to be the *same* list
+	// it does NOT have to be the *same* list. UnstructuredList returned from this fake client will NOT have apiVersion and kind set,
+	// but each Unstructured object in Items will preserve their respective apiVersion and kind. As a result, schema conversion for
+	// *List kinds may not work and conversion of each Unstructured object in Items may be required instead.
```

Comment:
*List kinds may not work

is it 'may not' work, or 'will not' work? does it sometimes work?

Reply:
If you are trying to convert UnstructuredList -> typed *List, and the UnstructuredList was retrieved by calling List(), it WILL not work because Kind will be empty. Was actually hoping someone with more experience in the fake dynamic client can confirm this since this is only based a quick scan I did of the code.

Reply:
I initially put "may not work" since it only applies when you call List from the fake dynamic client.

Reply:
Will not work.

I would also take a TODO or a PR to update the fake dynamic client to add a map of gvr to listKind and getKind.

Reply:
Opened a WIP PR for this https://github.com/kubernetes/kubernetes/pull/95951